src/main/java/io/codelex/flightplanner/FlightPlannerDatabaseService.java

```diff
@@ -0,0 +1,85 @@
+package io.codelex.flightplanner;
+
+import io.codelex.flightplanner.domain.Airport;
+import io.codelex.flightplanner.domain.Flight;
+import io.codelex.flightplanner.page.PageResult;
+import io.codelex.flightplanner.repository.AirportRepository;
+import io.codelex.flightplanner.repository.FlightRepository;
+import io.codelex.flightplanner.request.AddFlightRequest;
+import io.codelex.flightplanner.request.SearchFlightRequest;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.HttpStatusCode;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+
+public class FlightPlannerDatabaseService implements FlightPlannerService {
+
+    int currentId = 10;
+
+    private FlightRepository flightRepository;
+    private AirportRepository airportRepository;
+
+    public FlightPlannerDatabaseService(FlightRepository flightRepository, AirportRepository airportRepository) {
+        this.flightRepository = flightRepository;
+        this.airportRepository = airportRepository;
+    }
+
+    public synchronized Flight addFlight(AddFlightRequest request) {
+        int newId = currentId;
+        currentId++;
+        //Creating flight
+        Flight flight = new Flight(request.getFrom(), request.getTo(), request.getCarrier(), request.getDepartureTime(), request.getArrivalTime());
+        //Error handling
+        if (listFlights().stream().anyMatch(flight::equals)) {
+            throw new ResponseStatusException(HttpStatus.CONFLICT);
+        }
+        if (flight.hasSameAirports()) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST);
+        }
+        if (flight.getDepartureTime().isAfter(flight.getArrivalTime()) || flight.getDepartureTime().isEqual(flight.getArrivalTime())) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST);
+        }
+        //Adding and returning flight
+        flightRepository.save(flight);
+        return flight;
```

Comment:
Ja pareizi atceros, tad flightRepository.save metode atgriež saglabāto Flight, tāpēc korektāk būtu  `return flightRepository.save(flight);`